src/box/lua/init.c

```diff
@@ -574,20 +574,44 @@ box_lua_init(struct lua_State *L)
 	luamp_set_encode_extension(luamp_encode_extension_box);
 	luamp_set_decode_extension(luamp_decode_extension_box);
 
+	lua_getfield(L, LUA_REGISTRYINDEX, "_LOADED");
+
 	/* Load Lua extension */
-	for (const char **s = lua_sources; *s; s += 2) {
-		const char *modname = *s;
-		const char *modsrc = *(s + 1);
+	for (const char **s = lua_sources; *s; s += 3) {
+		const char *modfile_raw = *s;
+		const char *modname = *(s + 1);
+		const char *modsrc = *(s + 2);
```

Comment:
Why do you refuse to introduce a separate function for this also (like you've done in the previous patch)?

Reply:
The code in src/lua/init.c is used several times, so I moved it into a function. Here we need this code once, so I don't.

I think that I'll end up with a common function in a future, but there are small differences to investigate and eliminate (call/pcall and ones marked with TODO).

Reply:
That works for me, resolving then.